File: Heat/Models/Heater.vb

Namespace Models

    ''' <summary>
    ''' Rappresenta un bruciatore della caldaia; può essere l'unico bruciatore o uno dei tanti bruciatori.
    ''' </summary>
    ''' <remarks></remarks>
    Public Class Heater

        Property ID As Integer
        Property ThermalUnit As ThermalUnit

        Property Manifacturer As Manifacturer
        Property Model As ManifacturerModel
        Property SerialNumber As String

        Property MinimumPowerKW As Single
        Property MaximumPowerKW As Single

        Property InstallationDate As DateTime
        Property Type As String

        Property FuelID As Integer
        Property Fuel As Fuel

        Property DismissDate As Nullable(Of DateTime)

    End Class
End Namespace

File: Heat.ConvertedToC#/Models/Heater.cs
using System;

namespace Heat.Models
{

    /// <summary>
    /// Rappresenta un bruciatore della caldaia; può essere l'unico bruciatore o uno dei tanti bruciatori.
    /// </summary>
    /// <remarks></remarks>
    public class Heater
	{

		public int ID { get; set; }
		public ThermalUnit ThermalUnit { get; set; }

		public Manifacturer Manifacturer { get; set; }
		public ManifacturerModel Model { get; set; }
		public string SerialNumber { get; set; }

		public float MinimumPowerKW { get; set; }
		public float MaximumPowerKW { get; set; }

		public DateTime? InstallationDate { get; set; }
		public string Type { get; set; }

		public int FuelID { get; set; }
		public Fuel Fuel { get; set; }

		public Nullable<DateTime> DismissDate { get; set; }

	}
}

